monaco: properly pass `options` to `input`

vince-fugnitto opened the pull request:
<!--
Thank you for your Pull Request. Please provide a description and review
the requirements below.

Contributors guide: https://github.com/theia-ide/theia/blob/master/CONTRIBUTING.md

Note: Security vulnerabilities should not be disclosed on GitHub, through a PR or any
other means. See SECURITY.md at the root of this repository, to learn how to report
vulnerabilities.
-->

#### What it does
<!-- Include relevant issues and describe how they are addressed. -->

Fixes: #10094 

The pull-request addresses the issue where the `options` parameter used in `QuickInputService.input` was not being passed properly. This meant that configurations provided by callers were always disregarded.

#### How to test
<!-- Explain how a reviewer can reproduce a bug, test new functionality or verify performance improvements. -->

1. start the application
2. use <kbd>F1</kbd> to execute `Quick Input: Test Positive Integer`
3. confirm that options such as `placholder` and `validateInput` works correctly (positive integer allowed only)

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: vince-fugnitto <vincent.fugnitto@ericsson.com>

msujew commented on packages/monaco/src/browser/monaco-quick-input-service.ts:
```diff
@@ -182,8 +182,23 @@ export class MonacoQuickInputService implements QuickInputService {
-        return this.monacoService.input();
+        let inputOptions: monaco.quickInput.IInputOptions | undefined = undefined;
+        if (options) {
+            inputOptions = {
+                ignoreFocusLost: options?.ignoreFocusLost,
+                password: options?.password,
+                placeHolder: options?.placeHolder,
+                prompt: options?.prompt,
+                value: options?.value,
+                valueSelection: options?.valueSelection
+            };
```
Why not use the spread operator on the options here like `inputOptions = { ...options }`? Is there a benefit in manually assembling the options? Is there a special reason why we don't pass the options directly?

vince-fugnitto replied:
@msujew I could not use the spread since the `validateInput` typings are different for `InputOptions` and `IInputOptions`, and we'd get a compilation error.

I could however do the following, or perhaps align the typings:

```ts
input(options?: InputOptions, token?: CancellationToken): Promise<string | undefined> {
    let inputOptions: monaco.quickInput.IInputOptions | undefined = undefined;
    if (options) {
        let validateInput = undefined;
        if (options.validateInput) {
            validateInput = (input: string) => options!.validateInput!(input)!;
        }
        inputOptions = {...options, validateInput};
    }
    return this.monacoService.input(inputOptions, token);
}
```

msujew replied:
Alright I see, I didn't notice the difference in typing previously. That's alright with me 👍

msujew reviewed (commented):
The changes look good to me, the options are passed correct

vince-fugnitto commented:
The changes should be ready for an additional round of review when there's time :+1: 
- updated the api-sample to handle negative numbers
- updated to use a `spread` for simplicity

msujew reviewed (approved):
I can confirm that the issue exists on `master` and is addressed by the changes nicely:

- [x] Options are correctly passed to the input
- [x] Input validation is running successfully

One could argue that `0` is an element of the set of positive integers, but I don't think we need this discussion here 😄
